AG-12037 QA Feedback

olegat opened the pull request:
https://ag-grid.atlassian.net/browse/AG-12037

olegat committed:
AG-12037 QA Feedback: Fix TC 1

The color picker was actually getting shown correctly, but the
InteractionManager would also dispatch a 'click' event to series area,
which caused the color picker dialog to be closed immediately after.

alantreadway commented on packages/ag-charts-community/src/chart/interaction/interactionManager.ts:
```diff
@@ -282,11 +282,15 @@ export class InteractionManager extends InteractionStateListener<InteractionType
             return;
         }
+        // AG-12037 Interacting with HTML buttons can also fire events on the series, which we don't want.
+        if ([target?.tagName?.toLowerCase(), target?.role].includes('button')) {
```
We should probably just ensure to block event bubbling from within the overlays, either generically or for all specific points where we have consuming event handlers?

olegat replied:
Done.

olegat committed:
AG-12037 QA Feedback: Fix TC 2

Unfortunately, this has some duplicated logic with initMenuKeyNav to set
and reset the focus.

olegat committed:
AG-12037 CR Feedback: Fix TC 1

This fixes the bug by blocking event bubbling from within the overlay.

olegat committed:
AG-12037 Fix bug with focus-reset on popdowns

We need to reset the focus to `lastFocus` when the user pressed a menu
item with the keyboard. `initMenuKeyNav` does this for when the ESC key
is pressed, but not for when another key is pressed.

olegat committed:
AG-12037 Clear state in MenuCloser.finishClosing()

This avoid invoking the destroyFns and lastFocus.focus() functions
twice, which could potentionally have undesired effects.

olegat committed:
AG-12037 Fix e2e tests

This fixes these tests:

toolbar.spec.ts:
x toolbar > text
x toolbar > callout

The bug was that textualStateUtils.ts relies on event bubbling to
receive 'keydown' events from the InteractionManager to handle the
Escape and Enter key presses. Therefore, allow bubbling for 'keydown'
events. This may have some unintended consequences.

lsjroberts reviewed (approved):
LGTM
